src/main/java/com/linecorp/armeria/common/SerializationFormat.java

```diff
@@ -62,6 +74,15 @@
     public static Set<SerializationFormat> ofThrift() {
         return THRIFT_FORMAT;
     }
+
+    /**
+     * Returns the serialization format corresponding to the passed in {@code mimeType}, or
+     * {@link Optional#empty} if the mimetype is not recognized. {@code null} is treated as an unknown
+     * mimetype.
+     */
+    public static Optional<SerializationFormat> fromMimeType(@Nullable String mimeType) {
+        return Optional.ofNullable(MIME_TYPE_TO_FORMAT.get(mimeType));
```

Comment:
Yeah I was wondering how lenient this should be. An exact match is sufficient for armeria clients, but I guess a bit of looseness can help. Instead of a regex parse, I went with what seemed simpler to me, a normalization method, let me know what you think.